lightning/src/ln/shutdown_tests.rs

```diff
@@ -1209,3 +1210,81 @@ fn simple_target_feerate_shutdown() {
 	check_closed_event!(nodes[0], 1, ClosureReason::CooperativeClosure, [nodes[1].node.get_our_node_id()], 100000);
 	check_closed_event!(nodes[1], 1, ClosureReason::CooperativeClosure, [nodes[0].node.get_our_node_id()], 100000);
 }
+
+
+#[test]
+fn htlc_removal_no_early_closing_signed() {
+	// Previously, if we have a pending inbound HTLC on a channel which has initiated shutdown,
```

Comment:
I think the test coverage should also encompass a “pending outbound HTLC” to check if final HTLC removal works well in both directions. And current test is `send_payment(&nodes[0], &[&nodes[1]])`, so it’s more an outbound HTLC from the viewpoint of `nodes[0]` ? Nodes 1 receives the `SendClosingSigned` L1277

Reply:
Yes the test coverage of the other HTLC in `shutdown_tests.rs` would have to be test, though note I still think you’re testing the pending outbound HTLC here?